okhttp-tls/src/main/kotlin/okhttp3/tls/HeldCertificate.kt

```diff
@@ -516,7 +516,25 @@ class HeldCertificate(
       return HeldCertificate(keyPair, certificate)
     }
 
-    private fun decodePem(pem: String): X509Certificate {
+    /**
+     * Decodes a multiline string that contains a [certificate][certificatePem] which is
+     * [PEM-encoded][rfc_7468]. A typical input string looks like this:
+     *
+     * ```
+     * -----BEGIN CERTIFICATE-----
+     * MIIBYTCCAQegAwIBAgIBKjAKBggqhkjOPQQDAjApMRQwEgYDVQQLEwtlbmdpbmVl
+     * cmluZzERMA8GA1UEAxMIY2FzaC5hcHAwHhcNNzAwMTAxMDAwMDA1WhcNNzAwMTAx
+     * MDAwMDEwWjApMRQwEgYDVQQLEwtlbmdpbmVlcmluZzERMA8GA1UEAxMIY2FzaC5h
+     * cHAwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAASda8ChkQXxGELnrV/oBnIAx3dD
+     * ocUOJfdz4pOJTP6dVQB9U3UBiW5uSX/MoOD0LL5zG3bVyL3Y6pDwKuYvfLNhoyAw
+     * HjAcBgNVHREBAf8EEjAQhwQBAQEBgghjYXNoLmFwcDAKBggqhkjOPQQDAgNIADBF
+     * AiAyHHg1N6YDDQiY920+cnI5XSZwEGhAtb9PYWO8bLmkcQIhAI2CfEZf3V/obmdT
+     * yyaoEufLKVXhrTQhRfodTeigi4RX
+     * -----END CERTIFICATE-----
+     * ```
+     */
+    @JvmStatic
+    fun decodeCertificate(pem: String): X509Certificate {
```

Comment:
ooooh neat

Reply:
rename to `decodeCertificatePem` ?

Should this be an extension on String?
https://publicobject.com/2019/05/22/factory-or-extension-function/

Reply:
Also I don’t think this is the right file for this method. Should we have a file for certificates?

Reply:
HeldCertificate is intended to model certificates where we have a private key. Mabybe umm `object Certifcates { ... }` ?